pool/src/lib.rs

```diff
@@ -954,8 +996,11 @@ impl LendingPool {
         debt_token: &DebtTokenClient,
         amount_to_borrow: i128,
     ) -> Result<(), Error> {
-        let s_token_balance = s_token.balance(&who);
-        assert_with_error!(env, s_token_balance == 0, Error::MustNotBeInCollateralAsset);
+        assert_with_error!(
+            env,
+            !user_config.is_using_as_collateral(env, reserve.get_id()),
```

Comment:
What if user deposited some amount of asset and set collateral to false. Should we let them to borrow this asset in this case (if they have s_tokens)?